MachineLearningPrototype/utils/preprocess_data.py

```diff
@@ -15,3 +16,41 @@ def extract_quotation_content(content):
     if (len(substring) >= 1):
         result=' '.join(substring)
         return result
+
+def query_trss_for_jenkins_output(jenkins_url, test_name, trss_servers=["https://trss.adoptium.net"]):
+    #Get build info from TRSS
+    query_url = "https://trss.adoptium.net/api/parseJenkinsUrl"
+    query_params = {"jenkinsUrl": jenkins_url}
+    output = requests.get(query_url, query_params)
+    output_dict = json.loads(output.content)
+
+    if "output" not in output_dict:
+        raise Exception("Invalid jenkins Url")
+        return ""
+    
+    output_dict = output_dict["output"]
+
+    if output_dict["errorMsg"]:
+        raise Exception("Failed to parse Jenkins Url with error message: " + output_dict["errorMsg"])
+        return ""
+
+    url = output_dict["serverUrl"]
+    build_name = output_dict["buildName"]
+    build_num = output_dict["buildNum"]
+
+    for server in trss_servers:
+        #Try to fetch test data from server
+        query_url = f"{server}/api/getOutputByTestInfo"
+        query_params = {"url": url,
+                        "buildName": build_name,
+                        "buildNum": build_num,
+                        "testName": test_name,}
```

Comment:
`query_params` should not be defined inside the for loop. The value does not change. It should be defined outside the for loop.

Reply:
I don't know how I did not see this, thank you so much! Updated.